src/modules/settings-info/settings-info.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Users } from 'src/modules/entities/users.entity';
+import { Repository } from 'typeorm';
+import { CreateSettingsInfoDto } from './dto/create-settings-info.dto';
+import { UpdateSettingsInfoDto } from './dto/update-settings-info.dto';
+
+@Injectable()
+export class SettingsInfoService {
+  constructor(
+    @InjectRepository(Users)
+    private readonly settingsInfoRepo: Repository<Users>,
+  ) {}
+
+  async create(createSettingsInfoDto: CreateSettingsInfoDto) {
+    const { email } = createSettingsInfoDto;
+
+    const currentUser = await this.settingsInfoRepo.findOneBy({ email: email });
+
+    if (!currentUser) throw new NotFoundException();
+
+    return await this.settingsInfoRepo.save({
+      ...currentUser,
+      ...createSettingsInfoDto,
+    });
+  }
+
+  async findAll() {
+    return await this.settingsInfoRepo.find();
+  }
+
+  async findOne(id: number) {
+    return await this.settingsInfoRepo.findOneBy({ id: id });
+  }
+
+  async update(id: number, updateSettingsInfoDto: UpdateSettingsInfoDto) {
+    const currentSettings = await this.findOne(id);
+
+    if (!currentSettings) throw new NotFoundException();
+
+    return await this.settingsInfoRepo.save({
+      ...currentSettings,
```

Comment:
why we need add currentSettings when we save?

Reply:
It's because I don't want to change the password, role and googleID fields in the Users table. Since a user already needs to be created when the frontend accesses these routes, in the POST and PATCH requests this approach makes sure that we're only changing first name, last name and phone columns

Reply:
yes, but for it you should not pass existing data, you should pass only data that you want to update.

Reply:
Ok, I will find a way and change it